docs/developers/java-dev.rst

```diff
@@ -0,0 +1,64 @@
+Java Devolopment
+================
+
+Introduction
+------------
+
+GraphAr Java library based on GraphAr C++ library and an efficient FFI
+for Java and C++ called
+`FastFFI <https://github.com/alibaba/fastFFI>`__.
+
+Source Code Level
+~~~~~~~~~~~~~~~~~
+
+-  Interface
+
+-  Class
+
+-  JNI code
+
+-  C++ source code
+
+Developers only need to write interfaces with annotations. For
+annotation's usage, please refer to
+`FastFFI <https://github.com/alibaba/fastFFI>`__.
+
+FastFFI will automatically generate ``.java`` files which implement
+interfaces, and automatically generate ``.cc`` files which include JNI
+code for native methods. Obviously, C++ code has been written.
+
+If llvm4jni is opened, part of JNI code will be transferred to java
+class.
+
+Runtime Level
+~~~~~~~~~~~~~
+
+Interfaces and classes will be compiled to ``.class`` files.
+
+By writing CMakeLists.txt, all C++ dependents(e.g. JNI code, GraphAr C++
+library and other C++ library) will been integrated into a bridge
+dynamic library called gar-jni which can be called by native methods
+directly.
```

Comment:
Here can add a link to CMakeLists.txt as example

Reply:
We have written CMakeLists.txt, developers only need know the architecture, not need to write new CMakeLists.

I think add a link to CMakeLists is enough.

Reply:
If developer do not need to know the CMakeLists, that we should talk about CMakeList here